### DISCUSSION TITLE
nordigen import fails with error Undefined array key "liabilities"

### DISCUSSION THREAD

**Discussion by Alpoy** [firefly-iii/firefly-iii#6935] [Firefly III] [answered]
Hi,

I've tried multiple times importing from Nordigen (norwegian bank) but always get the error message : 
_Undefined array key "liabilities"_

![bilde](https://user-images.githubusercontent.com/11949996/214836703-3b68efdb-e12a-428b-a243-c033f32a725b.png)

Am I doing something wrong? I've followed all guides on Nordigen and testing the API requests through curl works just fine. 
Firefly is stock install, only settings (in the UI) I've changed are locale settings like language and currency. 
Data-importer 1.0.0 , Firefly: 5.7.18

Stack trace:

>   ErrorException:
> Undefined array key "liabilities"
> 
>   at /var/www/data-importer/app/Support/Internal/MergesAccountLists.php:165
>   at Illuminate\Foundation\Bootstrap\HandleExceptions->handleError()
>      (/var/www/data-importer/vendor/laravel/framework/src/Illuminate/Foundation/Bootstrap/HandleExceptions.php:266)
>   at Illuminate\Foundation\Bootstrap\HandleExceptions->Illuminate\Foundation\Bootstrap\{closure}()
>      (/var/www/data-importer/app/Support/Internal/MergesAccountLists.php:165)
>   at App\Http\Controllers\Import\ConfigurationController->mergeGenericAccountList()
>      (/var/www/data-importer/app/Support/Internal/MergesAccountLists.php:45)
>   at App\Http\Controllers\Import\ConfigurationController->mergeNordigenAccountLists()
>      (/var/www/data-importer/app/Http/Controllers/Import/ConfigurationController.php:120)
>   at App\Http\Controllers\Import\ConfigurationController->index()
>      (/var/www/data-importer/vendor/laravel/framework/src/Illuminate/Routing/Controller.php:54)
>   at Illuminate\Routing\Controller->callAction()
>      (/var/www/data-importer/vendor/laravel/framework/src/Illuminate/Routing/ControllerDispatcher.php:43)
>   at Illuminate\Routing\ControllerDispatcher->dispatch()
>      (/var/www/data-importer/vendor/laravel/framework/src/Illuminate/Routing/Route.php:259)
>   at Illuminate\Routing\Route->runController()
>      (/var/www/data-importer/vendor/laravel/framework/src/Illuminate/Routing/Route.php:205)
>   at Illuminate\Routing\Route->run()
>      (/var/www/data-importer/vendor/laravel/framework/src/Illuminate/Routing/Router.php:798)
>   at Illuminate\Routing\Router->Illuminate\Routing\{closure}()
>      (/var/www/data-importer/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:141)
>   at Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
>      (/var/www/data-importer/app/Http/Middleware/IsReadyForStep.php:54)
>   at App\Http\Middleware\ConfigurationControllerMiddleware->handle()
>      (/var/www/data-importer/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:180)
>   at Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
>      (/var/www/data-importer/vendor/laravel/framework/src/Illuminate/Routing/Middleware/SubstituteBindings.php:50)
>   at Illuminate\Routing\Middleware\SubstituteBindings->handle()
>      (/var/www/data-importer/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:180)
>   at Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
>      (/var/www/data-importer/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/VerifyCsrfToken.php:78)
>   at Illuminate\Foundation\Http\Middleware\VerifyCsrfToken->handle()
>      (/var/www/data-importer/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:180)
>   at Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
>      (/var/www/data-importer/vendor/laravel/framework/src/Illuminate/View/Middleware/ShareErrorsFromSession.php:49)
>   at Illuminate\View\Middleware\ShareErrorsFromSession->handle()
>      (/var/www/data-importer/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:180)
>   at Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
>      (/var/www/data-importer/vendor/laravel/framework/src/Illuminate/Session/Middleware/StartSession.php:121)
>   at Illuminate\Session\Middleware\StartSession->handleStatefulRequest()
>      (/var/www/data-importer/vendor/laravel/framework/src/Illuminate/Session/Middleware/StartSession.php:64)
>   at Illuminate\Session\Middleware\StartSession->handle()
>      (/var/www/data-importer/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:180)
>   at Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
>      (/var/www/data-importer/vendor/laravel/framework/src/Illuminate/Cookie/Middleware/AddQueuedCookiesToResponse.php:37)
>   at Illuminate\Cookie\Middleware\AddQueuedCookiesToResponse->handle()
>      (/var/www/data-importer/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:180)
>   at Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
>      (/var/www/data-importer/vendor/laravel/framework/src/Illuminate/Cookie/Middleware/EncryptCookies.php:67)
>   at Illuminate\Cookie\Middleware\EncryptCookies->handle()
>      (/var/www/data-importer/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:180)
>   at Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
>      (/var/www/data-importer/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:116)
>   at Illuminate\Pipeline\Pipeline->then()
>      (/var/www/data-importer/vendor/laravel/framework/src/Illuminate/Routing/Router.php:797)
>   at Illuminate\Routing\Router->runRouteWithinStack()
>      (/var/www/data-importer/vendor/laravel/framework/src/Illuminate/Routing/Router.php:776)
>   at Illuminate\Routing\Router->runRoute()
>      (/var/www/data-importer/vendor/laravel/framework/src/Illuminate/Routing/Router.php:740)
>   at Illuminate\Routing\Router->dispatchToRoute()
>      (/var/www/data-importer/vendor/laravel/framework/src/Illuminate/Routing/Router.php:729)
>   at Illuminate\Routing\Router->dispatch()
>      (/var/www/data-importer/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php:190)
>   at Illuminate\Foundation\Http\Kernel->Illuminate\Foundation\Http\{closure}()
>      (/var/www/data-importer/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:141)
>   at Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
>      (/var/www/data-importer/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TransformsRequest.php:21)
>   at Illuminate\Foundation\Http\Middleware\TransformsRequest->handle()
>      (/var/www/data-importer/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/ConvertEmptyStringsToNull.php:31)
>   at Illuminate\Foundation\Http\Middleware\ConvertEmptyStringsToNull->handle()
>      (/var/www/data-importer/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:180)
>   at Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
>      (/var/www/data-importer/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TransformsRequest.php:21)
>   at Illuminate\Foundation\Http\Middleware\TransformsRequest->handle()
>      (/var/www/data-importer/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TrimStrings.php:40)
>   at Illuminate\Foundation\Http\Middleware\TrimStrings->handle()
>      (/var/www/data-importer/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:180)
>   at Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
>      (/var/www/data-importer/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/ValidatePostSize.php:27)
>   at Illuminate\Foundation\Http\Middleware\ValidatePostSize->handle()
>      (/var/www/data-importer/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:180)
>   at Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
>      (/var/www/data-importer/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/PreventRequestsDuringMaintenance.php:86)
>   at Illuminate\Foundation\Http\Middleware\PreventRequestsDuringMaintenance->handle()
>      (/var/www/data-importer/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:180)
>   at Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
>      (/var/www/data-importer/vendor/fruitcake/laravel-cors/src/HandleCors.php:38)
>   at Fruitcake\Cors\HandleCors->handle()
>      (/var/www/data-importer/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:180)
>   at Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
>      (/var/www/data-importer/vendor/laravel/framework/src/Illuminate/Http/Middleware/TrustProxies.php:39)
>   at Illuminate\Http\Middleware\TrustProxies->handle()
>      (/var/www/data-importer/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:180)
>   at Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
>      (/var/www/data-importer/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:116)
>   at Illuminate\Pipeline\Pipeline->then()
>      (/var/www/data-importer/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php:165)
>   at Illuminate\Foundation\Http\Kernel->sendRequestThroughRouter()
>      (/var/www/data-importer/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php:134)
>   at Illuminate\Foundation\Http\Kernel->handle()
>      (/var/www/data-importer/public/index.php:76)

**Comment by JC5** [comment, marked as answer]
That's probably a bug, thanks for letting me know! v1.0.1 just got out but I don't expect that will fix it.

**Comment by JC5** [comment]
v1.0.2 will fix this :+1:
